electron-builder.config.js

```diff
@@ -0,0 +1,145 @@
+const path = require("path");
+const fs = require("fs");
+
+const VOICEVOX_ENGINE_DIR =
+  process.env.VOICEVOX_ENGINE_DIR ?? "../voicevox_engine/run.dist/";
+
+// ${productName} Web Setup ${version}.${ext}
+const NSIS_WEB_ARTIFACT_NAME = process.env.NSIS_WEB_ARTIFACT_NAME;
+
+// ${productName}-${version}.${ext}
+const LINUX_ARTIFACT_NAME = process.env.LINUX_ARTIFACT_NAME;
+
+// ${packageName}
+const LINUX_EXECUTABLE_NAME = process.env.LINUX_EXECUTABLE_NAME;
+
+// ${productName}-${version}.${ext}
+const MACOS_ARTIFACT_NAME = process.env.MACOS_ARTIFACT_NAME;
+
+const isMac = process.platform === "darwin";
+
+/** @type {import("electron-builder").Configuration} */
+const builderOptions = {
+  beforeBuild: async () => {
+    if (fs.existsSync(path.resolve(__dirname, "dist_electron"))) {
+      fs.rmSync(path.resolve(__dirname, "dist_electron"), { recursive: true });
+    }
+  },
+  directories: {
+    output: "dist_electron",
+    buildResources: "build",
+  },
+  files: ["dist/**/*", "package.json"],
+  fileAssociations: [
+    {
+      ext: "vvproj",
+      name: "VOICEVOX Project file",
+      description: "VOICEVOX Project file",
+      role: "Editor",
+      icon: "icons/vvproj." + (isMac ? "icns" : "ico"),
+    },
+    {
+      ext: "vvpp",
+      name: "VOICEVOX Plugin package",
+      description: "VOICEVOX Plugin package",
+      role: "Editor",
+      icon: "icons/vvpp." + (isMac ? "icns" : "ico"),
+    },
+    {
+      ext: "vvppp",
+      name: "VOICEVOX Plugin package (part)",
+      description: "VOICEVOX Plugin package (part)",
+      role: "Editor",
+      icon: "icons/vvpp." + (isMac ? "icns" : "ico"),
+    },
+  ],
+  extraFiles:
+    // electron-builderのextraFilesは、ファイルのコピー先としてVOICEVOX.app/Contents/を使用する。
+    // しかし、実行ファイルはVOICEVOX.app/Contents/MacOS/にあるため、extraFilesをVOICEVOX.app/Contents/ディレクトリにコピーするのは正しくない。
+    // VOICEVOX.app/Contents/MacOS/ディレクトリにコピーされるように修正する。
+    // cf: https://k-hyoda.hatenablog.com/entry/2021/10/23/000349#%E8%BF%BD%E5%8A%A0%E5%B1%95%E9%96%8B%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB%E5%85%88%E3%81%AE%E8%A8%AD%E5%AE%9A
+
+    isMac
+      ? [
+          { from: "build/README.txt", to: "MacOS/README.txt" },
+          {
+            from: ".env.production",
+            to: "MacOS/.env",
+          },
+          {
+            from: VOICEVOX_ENGINE_DIR,
+            to: "MacOS/",
+          },
+        ]
+      : [
+          { from: "build/README.txt", to: "README.txt" },
+          {
+            from: ".env.production",
+            to: ".env",
+          },
+          {
+            from: VOICEVOX_ENGINE_DIR,
+            to: "",
+          },
```

Comment:
どこかに`const pathPrefix = isMac ? "MacOS/" : ""`などを定義しておき、

```suggestion
    [
      { from: "build/README.txt", to: `${pathPrefix}README.txt` },
      {
        from: ".env.production",
        to: `${pathPrefix}.env`,
      },
      {
        from: VOICEVOX_ENGINE_DIR,
        to: `${pathPrefix}`,
      },
    ]
```

とする方がコンパクトになって綺麗かなと思いました！

Reply:
プレフィックスみたいにするようにしました。